snappy/SymServer_RequestHandler.py

```diff
@@ -22,33 +23,44 @@ def sendHeaders(self, code):
         self.set_status(code)
         self.set_header("Content-type", "application/json")
 
-    def head(self):
+    def head(self, path):
         uri = self.request.uri
         self.log(logLevel.INFO, "Cannot process HEAD request: {}".format(uri))
         self.sendHeaders(405)
 
-    def get(self):
-        uri = self.request.uri
-        self.log(logLevel.INFO, "Cannot process GET request: {}".format(uri))
-        self.sendHeaders(405)
+    @tornado.gen.coroutine
+    def get(self, path):
+        if path == "/__lbheartbeat__":
+            self.sendHeaders(200)
+            return
+        elif path == "/__heartbeat__":
+            # Translate this to a debug request, which the server already has a
+            # mechanism for handling. Then feed it in as if it was a POST
+            # request.
+            self.debugAllowed = True
```

Comment:
You are misunderstanding how request handlers work in Tornado. An instance of the request handler is made for each request of the server. Setting debugAllowed to true allows that request ONLY to use the debug protocol